ipa-core/src/protocol/basics/check_zero.rs

```diff
@@ -48,26 +55,82 @@ where
 
     let rv_share =
         semi_honest_multiply(ctx.narrow(&Step::MultiplyWithR), record_id, &r_sharing, v).await?;
+    tracing::info!("{:?}", &rv_share);
     let rv = F::from_array(
-        &rv_share
-            .reveal(ctx.narrow(&Step::RevealR), record_id)
-            .await?,
+        &malicious_reveal(ctx.narrow(&Step::RevealR), record_id, None, &rv_share)
+            .await?
+            .expect("full reveal should always return a value"),
     );
 
+    tracing::info!("{:?}", &rv);
     Ok(rv == F::ZERO)
 }
 
+#[cfg(test)]
+pub async fn check_zero_fp32bitprime<C>(
```

Comment:
do you have plans to use it outside of this module or we can move it into `tests`?

Reply:
the same argument applies to `SH1` as well